docs/source/config_files.rst

```diff
@@ -0,0 +1,105 @@
+trying the other configuration files before deciding to use your own Dockerfile.
+
+See the `Binder Documentation <https://mybinder.readthedocs.io/en/latest/dockerfile.html>`_ for
+best-practices with Dockerfiles.
+
+``environment.yml``
+^^^^^^^^^^^^^^^^^^^
+
+This is a conda environment specification, that lets you install packages with conda.
+You can also install files from pip in your ``environment.yml`` as well.
+Our example `enviornment.yml <https://github.com/binder-examples/python-conda_pip/blob/master/environment.yml>`_
+shows how one can specify a conda environment for repo2docker.
+
+``requirements.txt``
+^^^^^^^^^^^^^^^^^^^^
+
+This specifies a list of Python packages that should be installed in your
+ environment. Our `requirements.txt example <https://github.com/binder-examples/requirements/blob/master/requirements.txt>`_
+on GitHub shows a typical requirements file.
+
+``REQUIRE``
+^^^^^^^^^^^
+
+This specifies a list of Julia packages. Repositories with a  ``REQUIRE`` file
+**must also contain an** ``environment.yml`` **file**.  To see an example of a
+Julia repository with ``REQUIRE`` and ``environment.yml``,
+visit `binder-examples/julia-python <https://github.com/binder-examples/julia-python>`_.
+
+``install.R``
+^^^^^^^^^^^^^
+
+This is used to install R libraries pinned to a specific snapshot on
+`MRAN <https://mran.microsoft.com/documents/rro/reproducibility>`_.
+To set the date of the snapshot add a runtime.txt_.
+For an example ``install.R`` file, visit our `example install.R file <https://github.com/binder-examples/r/blob/master/install.R>`_.
+
+``apt.txt``
+^^^^^^^^^^^
+
+A list of Debian packages that should be installed. The base image used is usually the latest released
+version of Ubuntu.
+
+We use ``apt.txt``, for example, to install LaTeX in our
+`example apt.txt for LaTeX <https://github.com/binder-examples/latex/blob/master/apt.txt>`_.
+
+
+``setup.py``
+^^^^^^^^^^^^
+
+To install your repository like a Python package, you may include a
+``setup.py`` file. repo2docker installs ``setup.py`` files by running
+``pip install -e .``.
+While one can specify dependencies in ``setup.py``,
+repo2docker **requires configuration files such as** ``environment.yml`` or
+``requirements.txt``
+
+``postBuild``
+^^^^^^^^^^^^^
+
+A script that can contain arbitrary commands to be run after the whole repository has been built. If you
+want this to be a shell script, make sure the first line is ```#!/bin/bash``.
+
+An example usecase of ``postBuild`` file is JupyterLab's demo on mybinder.org.
+It uses a ``postBuild`` file in a folder called ``binder`` to `prepare
+their demo for binder <https://github.com/jupyterlab/jupyterlab-demo/blob/master/binder/postBuild>`_.
+
+.. _runtime.txt:
+
+``runtime.txt``
+^^^^^^^^^^^^^^^
+
+This allows you to control the runtime of Python or R.
+
+Adding ``python-2.7`` in the file our repository will run in a virtualenv with
```

Comment:
iTo use python-2.7: add python-2.7 in runtime.txt file. The repository will run ...